x/evm/types/hooks.go

```diff
@@ -0,0 +1,96 @@
+package types
+
+import (
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	ethcmn "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+var (
+	// BankSendEvent represent the signature of
+	// `event __CosmosNativeBankSend(address recipient, uint256 amount, string denom)`
+	BankSendEvent abi.Event
+
+	_ EvmHooks = MultiEvmHooks{}
+	_ EvmHooks = BankSendHook{}
+)
+
+// MultiEvmHooks combine multiple governance hooks, all hook functions are run in array sequence
+type MultiEvmHooks []EvmHooks
+
+// NewMultiEvmHooks combine multiple governance hooks
+func NewMultiEvmHooks(hooks ...EvmHooks) MultiEvmHooks {
+	return hooks
+}
+
+// PostTxProcessing delegate the call to underlying hooks
+func (h MultiEvmHooks) PostTxProcessing(ctx sdk.Context, txHash ethcmn.Hash, logs []*ethtypes.Log) error {
+	for i := range h {
+		err := h[i].PostTxProcessing(ctx, txHash, logs)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func init() {
+	addressType, _ := abi.NewType("address", "", nil)
+	uint256Type, _ := abi.NewType("uint256", "", nil)
+	stringType, _ := abi.NewType("string", "", nil)
+	BankSendEvent = abi.NewEvent(
+		"__CosmosNativeBankSend",
+		"__CosmosNativeBankSend",
+		false,
+		abi.Arguments{abi.Argument{
+			Name:    "recipient",
+			Type:    addressType,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "amount",
+			Type:    uint256Type,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "denom",
+			Type:    stringType,
+			Indexed: false,
+		}},
+	)
+}
+
+// BankSendHook handle the `__CosmosNativeBankSend` log.
+type BankSendHook struct {
```

Comment:
the BankSendHook shouldn't be included in this PR or module

Reply:
done